### PR TITLE
add dcompact layout for minivan

### PR DESCRIPTION
**Pull request by loksonarius**
## Description

Implementing my layout for the minivan keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/thevankeyboards/minivan/keymaps/dcompact/config.h [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#pragma once
```
Comment (drashna) [inline keyboards/thevankeyboards/minivan/keymaps/dcompact/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/thevankeyboards/minivan/keymaps/dcompact/keymap.c [hunk 2]
```diff
@@ -0,0 +1,98 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/thevankeyboards/minivan/keymaps/dcompact/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/thevankeyboards/minivan/keymaps/dcompact/keymap.c [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+#define AM_RGH A(S(KC_L))
+#define AM_CW A(S(KC_K))
+#define AM_CCW A(S(KC_J))
+#define AM_TLL A(S(KC_1))
+#define AM_TLW A(S(KC_2))
+#define AM_BSP A(S(KC_3))
+#define AM_FUL A(S(KC_4))
+
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/thevankeyboards/minivan/keymaps/dcompact/keymap.c:28]: This is not needed and should be removed. 
```suggestion
```

File: keyboards/thevankeyboards/minivan/keymaps/dcompact/rules.mk [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# https://beta.docs.qmk.fm/reference/config-options#feature-options
```
Comment (drashna) [inline keyboards/thevankeyboards/minivan/keymaps/dcompact/rules.mk:1]:
```suggestion
# https://docs.qmk.fm/#/config_options?id=feature-options
```

**Review by drashna — CHANGES_REQUESTED** [review]
n

**Comment by loksonarius** [comment]
Updated with GPL headers and other suggestions.
Thank you so much for the feedback -- I'll keep the suggestions in mind for future updates 🙏

File: keyboards/thevankeyboards/minivan/keymaps/dcompact/keymap.c [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  _RAISE,
+  _ADJUST
+};
+
+enum keycodes {
+  BASE = SAFE_RANGE,
+  LOWER,
+  RAISE,
+  ADJUST
+};
```
Comment (zvecr) [inline keyboards/thevankeyboards/minivan/keymaps/dcompact/keymap.c:57]:
```suggestion
#define LOWER MO(_LOWER)
#define RAISE MO(_RAISE)
```

File: keyboards/thevankeyboards/minivan/keymaps/dcompact/keymap.c [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      } else {
+        layer_off(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      }
+      return false;
+      break;
+  }
+  return true;
+};
```
Comment (zvecr) [inline keyboards/thevankeyboards/minivan/keymaps/dcompact/keymap.c:113]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
}
```

File: keyboards/thevankeyboards/minivan/keymaps/dcompact/config.h [hunk 7]
```diff
@@ -0,0 +1,17 @@
+/* Copyright 2020 Dan Herrera
```
Comment (zvecr) [inline keyboards/thevankeyboards/minivan/keymaps/dcompact/config.h:1]: This file can be removed, if there is no intention to add any config overrides.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/thevankeyboards/minivan/keymaps/dcompact/config.h (hunk 7 above)
Comment (loksonarius) [reply keyboards/thevankeyboards/minivan/keymaps/dcompact/config.h]: Done 👍

**Comment by loksonarius** [comment]
Thanks for the feedback @zvecr, I appreciate it 🙏

**Review by zvecr — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by Erovia — APPROVED** [review]

**Pull request merged**